bundle/Installer/NetgenRemoteMediaSiteInstaller.php

```diff
@@ -0,0 +1,157 @@
+                    static function (Repository $repository) use ($contentType): ContentTypeDraft {
+                        return $repository->getContentTypeService()->createContentTypeDraft($contentType);
+                    }
+                );
+
+                foreach ($fieldsToMigrate as $fieldDefinition) {
+                    $this->migrateField($fieldDefinition, $contentTypeDraft);
+                }
+
+                $this->repository->sudo(
+                    static function (Repository $repository) use ($contentTypeDraft): void {
+                        $repository->getContentTypeService()->publishContentTypeDraft($contentTypeDraft);
+                    }
+                );
+
+                $this->repository->commit();
+            } catch (Throwable $e) {
+                $this->repository->rollback();
+
+                throw $e;
+            }
+        }
+    }
+
+    /**
+     * @return \eZ\Publish\API\Repository\Values\ContentType\ContentType[]
+     */
+    private function getContentTypes(): array
+    {
+        $contentTypeGroups = $this->contentTypeService->loadContentTypeGroups();
+
+        $contentTypes = [];
+        foreach ($contentTypeGroups as $contentTypeGroup) {
+            $contentTypes = array_merge($contentTypes, $this->contentTypeService->loadContentTypes($contentTypeGroup));
+        }
+
+        return $contentTypes;
+    }
+
+    /**
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition[]
+     */
+    private function getFieldsToMigrate(ContentType $contentType): array
+    {
+        $fields = $contentType->getFieldDefinitions();
+        $fieldsToMigrate = [];
+        foreach ($fields as $field) {
+            if (
+                in_array($field->fieldTypeIdentifier, $this->fieldTypesToMigrate, true)
+                && !in_array($field->identifier, $this->excludedFieldIdentifiers, true)
+            ) {
+                $fieldsToMigrate[] = $field;
+            }
+        }
+
+        return $fieldsToMigrate;
+    }
+
+    private function migrateField(FieldDefinition $fieldDefinition, ContentTypeDraft $contentTypeDraft): void
+    {
+        $this->repository->sudo(
+            static function (Repository $repository) use ($contentTypeDraft, $fieldDefinition): void {
+                $repository->getContentTypeService()->removeFieldDefinition(
+                    $contentTypeDraft,
+                    $fieldDefinition
+                );
+            }
+        );
+
+        $fieldDefinitionCreateStruct = $this->contentTypeService->newFieldDefinitionCreateStruct(
+            $fieldDefinition->identifier,
+            'ngremotemedia'
+        );
+        $fieldDefinitionCreateStruct->names = $fieldDefinition->getNames();
+        $fieldDefinitionCreateStruct->position = $fieldDefinition->position;
+
+        $this->repository->sudo(
+            static function (Repository $repository) use ($contentTypeDraft, $fieldDefinitionCreateStruct): void {
```

Comment:
You can use non-static closure and then use `$this->contentTypeService`, since it's already available, but this up to you, it is purely for readability.